lib/src/migrators/module.dart

```diff
@@ -870,23 +878,52 @@ class _ModuleMigrationVisitor extends MigrationVisitor {
       if (expectedUrl != url) continue;
 
       var newName = renamedMembers[declaration] ?? declaration.name;
-      if (declaration.member is VariableDeclaration) newName = "\$$newName";
+      String importOnlyPrefix;
+      if (declaration.isImportOnly && declaration.forward.prefix != null) {
+        importOnlyPrefix = declaration.forward.prefix;
+        newName = declaration.name.substring(importOnlyPrefix.length);
+      }
+      var formattedNewName =
+          declaration.member is VariableDeclaration ? '\$$newName' : newName;
       if (_shouldForward(declaration.name) &&
           !declaration.name.startsWith('-')) {
-        shown.add(newName);
-      } else if (!newName.startsWith('-') && !newName.startsWith(r'$-')) {
-        hidden.add(newName);
+        var subprefix = "";
+        if (prefixToRemove != null &&
+            declaration.name.startsWith(prefixToRemove) &&
+            importOnlyPrefix != null) {
+          subprefix = importOnlyPrefix.substring(prefixToRemove.length);
+        }
+        if (declaration.name.length != newName.length) _needsImportOnly = true;
```

Comment:
Why not `declaration.name != newName`?

Reply:
Fixed. I was worried about an issue with underscores and dashes, but I forgot that both of these names should already be normalized to dashes.